benchmark/sparse_blas/sparse_blas.cpp

```diff
@@ -0,0 +1,498 @@
+TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A
+PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+******************************<GINKGO LICENSE>*******************************/
+
+#include <ginkgo/ginkgo.hpp>
+
+
+#include <algorithm>
+#include <chrono>
+#include <cstdlib>
+#include <exception>
+#include <fstream>
+#include <functional>
+#include <iomanip>
+#include <iostream>
+#include <map>
+#include <numeric>
+#include <random>
+#include <typeinfo>
+#include <unordered_set>
+
+
+#include "benchmark/utils/general.hpp"
+#include "benchmark/utils/loggers.hpp"
+#include "benchmark/utils/spmv_common.hpp"
+#include "benchmark/utils/types.hpp"
+#include "core/test/utils/matrix_generator.hpp"
+
+
+const auto benchmark_name = "sparse_blas";
+
+
+using itype = gko::int32;
+using Mtx = gko::matrix::Csr<etype, itype>;
+using mat_data = gko::matrix_data<etype, itype>;
+
+
+const std::map<std::string,
+               const std::function<std::shared_ptr<Mtx::strategy_type>()>>
+    strategy_map{
+        {"classical", [] { return std::make_shared<Mtx::classical>(); }},
+        {"sparselib", [] { return std::make_shared<Mtx::sparselib>(); }}};
+
+DEFINE_string(operations, "spgemm,spgeam,transpose",
+              "Comma-separated list of operations to be benchmarked. Can be "
+              "spgemm, spgeam, transpose");
+
+DEFINE_string(strategies, "classical,sparselib",
+              "Comma-separated list of CSR strategies: classical, sparselib");
+
+DEFINE_int32(
+    spgeam_swap_distance, 100,
+    "Maximum distance for row swaps to avoid rows with disjoint column ranges");
+
+DEFINE_string(
+    spgemm_mode, "normal",
+    "Which matrix B should be used to compute A * B: normal, "
+    "transposed, sparse, dense\n"
+    "normal: B = A for A square, A^T otherwise\ntransposed: B = "
+    "A^T\nsparse: B is a sparse matrix with dimensions of A^T with uniformly "
+    "random values, at most -rowlength non-zeros per row\ndense: B is a "
+    "'dense' sparse matrix with -rowlength columns and non-zeros per row");
+
+DEFINE_int32(spgemm_rowlength, 10,
+             "The length of rows in randomly generated matrices B. Only "
+             "relevant for spgemm_mode = <sparse|dense>");
+
+DEFINE_bool(validate_results, false,
+            "Check for correct sparsity pattern and compute the L2 norm "
+            "against the ReferenceExecutor solution.");
+
+
+std::pair<bool, double> validate_result(const Mtx *correct_mtx,
```

Comment:
Maybe put some comments here or the output to describe the error meaning
false for inconsistent matrix structure and the true, double is for numerical error, right?